pandas/core/algorithms.py

```diff
@@ -929,10 +931,13 @@ def value_counts_internal(
 
             # For backwards compatibility, we let Index do its normal type
             #  inference, _except_ for if if infers from object to bool.
-            idx = Index(keys)
+            idx = Index(keys, dtype=input_dtype if input_dtype != "float16" else None)
```

Comment:
Should these changes not be here after your comments in https://github.com/pandas-dev/pandas/pull/56187?

Reply:
oh very good point, forgot the revert. thx. does